dipy/tracking/fbcmeasures.pyx

```diff
@@ -0,0 +1,334 @@
+        
+        # if the fibers are too short FBC measures cannot be applied, 
+        # remove these.
+        streamlines_length = np.array([len(x) for x in py_streamlines], 
+                                dtype=np.int32)
+        minLength = min(streamlines_length)
+        if minLength < 10:
+            print("The minimum fiber length is 10 points. \
+                    Shorter fibers were found and removed.")
+            py_streamlines = [x for x in py_streamlines if len(x) >= 10]
+            streamlines_length = np.array([len(x) for x in py_streamlines], 
+                                    dtype=np.int32)
+            minLength = min(streamlines_length)
+        numberOfFibers = len(py_streamlines)
+        self.streamline_length = streamlines_length
+        maxLength = max(streamlines_length)
+        
+        dim = 3
+
+        # get lookup table info
+        lut = kernel.get_lookup_table()
+        N = lut.shape[2]
+        hn = (N-1) / 2
+        
+        # prepare numpy arrays for speed
+        streamlines = np.zeros((numberOfFibers, maxLength, dim), 
+                                dtype=np.float64) * np.nan
+        streamlines_tangents = np.zeros((numberOfFibers, maxLength, dim), 
+                                dtype=np.float64)
+        streamlines_nearestp = np.zeros((numberOfFibers, maxLength), 
+                                dtype=np.int32)
+        streamline_scores = np.zeros((numberOfFibers, maxLength), 
+                                dtype=np.float64) * np.nan
+        
+        # copy python streamlines into numpy array
+        for lineId in range(numberOfFibers):
+            for pointId in range(streamlines_length[lineId]):
+                for dim in range(3):
+                    streamlines[lineId, pointId, dim] = \
+                        py_streamlines[lineId][pointId][dim]
+        self.streamline_points = streamlines
+        
+        # compute tangents
+        for lineId in range(numberOfFibers):
+            for pointId in range(streamlines_length[lineId] - 1):
+                tangent = np.subtract(streamlines[lineId, pointId + 1], 
+                                        streamlines[lineId, pointId])
+                streamlines_tangents[lineId, pointId] = np.divide(tangent, 
+                                            np.sqrt(np.dot(tangent, tangent)))
+        
+        # estimate which kernel LUT index corresponds to angles
+        tree = KDTree(kernel.get_orientations())
+        for lineId in range(numberOfFibers):
+            for pointId in range(streamlines_length[lineId] - 1):
+                streamlines_nearestp[lineId, pointId] = \
+                    tree.query(streamlines[lineId, pointId])[1]
+        
+        # arrays for parallel computing
+        score_mp = np.zeros(numberOfFibers)
+        xd_mp = np.zeros(numberOfFibers, dtype=np.int32)
+        yd_mp = np.zeros(numberOfFibers, dtype=np.int32)
+        zd_mp = np.zeros(numberOfFibers, dtype=np.int32)
```

Comment:
Again, I wonder whether these arrays need to be numpy, or can be allocated using cdef instead? I am really not sure, but thought I would ask.